From a44a678d823aba8e2911c0ce367b56e94b814d47 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Wed, 23 Sep 2015 16:07:10 +0200
Subject: [PATCH] source_registry_server_process_file_monitor_event - Correct
 possible runtime warning

When the source file is not loaded yet, then the corresponding ESource
object is NULL, but the call calls g_object_unref() on it, which leads
to a runtime warning, even harmless.
---
 libebackend/e-source-registry-server.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libebackend/e-source-registry-server.c b/libebackend/e-source-registry-server.c
index 7d34c66..4552c95 100644
--- a/libebackend/e-source-registry-server.c
+++ b/libebackend/e-source-registry-server.c
@@ -500,7 +500,7 @@ source_registry_server_process_file_monitor_event (gpointer key,
 			return;
 		}
 
-		g_object_unref (source);
+		g_clear_object (&source);
 	}
 
 	if (event_type == G_FILE_MONITOR_EVENT_CREATED) {
-- 
2.5.0

